src/impy/kinematics.py

```diff
@@ -86,7 +86,7 @@ class CompositeTarget(object):
 
     from numpy.random import default_rng
 
-    def __init__(self, component_list, label="", random_state=None):
+    def __init__(self, component_list, label="", random_state=None, nevents=None):
```

Comment:
The composite target should not get `nevents` via the constructor. This breaks the concept. The generator knows how many events to generate, not the EventKinematics object.